lib/smart_answer_flows/marriage-abroad.rb

```diff
@@ -197,6 +204,13 @@ def define
           elsif calculator.partner_is_same_sex?
             if %w(belgium norway).include?(calculator.ceremony_country)
               outcome :outcome_same_sex_civil_partnership_in_affirmation_countries
+            elsif calculator.ceremony_country == 'south-africa' &&
+                (
+                  calculator.resident_of_uk? ||
+                  calculator.resident_of_ceremony_country? ||
+                  calculator.resident_of_third_country?
+                )
+              outcome :outcome_opposite_same_sex_marriage_residing_in_uk_or_south_africa
```

Comment:
The checks for a partner of the opposite sex or same sex send you to the same outcome. I think you should consolidate these into one statement that just checks the value of the ceremony_country is 'south-africa'